storage/statedb/database.go

```diff
@@ -474,6 +487,13 @@ func (db *Database) Node(hash common.Hash) ([]byte, error) {
 	return enc, err
 }
 
+// DoesExistCachedNode return if the noce exist from cached trie node in memory.
+func (db *Database) DoesExistCachedNode(hash common.Hash) bool {
```

Comment:
Don't we need to look for the node in the trie node cache?
Also, it seems that you need to lock the `db.lock.RLock()` according to the line 475~477.

Reply:
I added Lock. Thanks. :)